infretis/core/tis.py

```diff
@@ -279,9 +276,25 @@ def select_shoot(
         "sh": shoot,
     }
 
+    # set engine, mdrun, rng, then clean_up
+    for ens_num in picked.keys():
+        pens = picked[ens_num]
+        if pens["ens"]["tis_set"]["quantis"] and ens_num == -1:
+            engine = ENGINES[-1]
+        elif pens["ens"]["tis_set"]["multi_engine"]:
+            engine = ENGINES[ens_num]
+        else:
+            engine = ENGINES[picked[ens_num]["pin"]]
+        engine.set_mdrun(pens)
+        if "rgen-eng" in pens:
+            engine.rgen = pens["rgen-eng"]
+        engine.clean_up()
+        pens["ens"]["engine"] = engine
```

Comment:
It was this line that created the bug. I am not sure why, but maybe adding the engine to the `pens` dict makes the engine get pickled.